knative-operator/pkg/common/openshift.go

```diff
@@ -71,6 +74,61 @@ func ingress(ks *servingv1alpha1.KnativeServing, c client.Client) error {
 	return nil
 }
 
+// Update updates Knative controller env to use cluster wide proxy information
+func Update(ks *servingv1alpha1.KnativeServing, c client.Client) error {
```

Comment:
I'm not sure this function should live here as this file for now only contains mutations for the KnativeServing resource itself. Should we create a new file that encapsulates mutations to the deployment as it runs? Do we even want to do that or do we want to enhance the KS API to allow us to set these proxy settings through them?

Aaaaand the function name is somewhat non-descriptive. `ApplyProxySettings` or something along the lines would be more self-explanatory.